verify.go

```diff
@@ -450,3 +451,120 @@ func verifyClaims(iat time.Time, exp time.Time, replid, user, cluster, subcluste
 
 	return verifyRawClaims(replid, user, cluster, subcluster, deployment, claims, false, false, false, false, false)
 }
+
+func decodeUnsafePASETO(token string) ([]byte, error) {
+	parts := strings.Split(token, ".")
+	if len(parts) != 4 {
+		return nil, fmt.Errorf("token not in PASETO format")
+	}
+	if parts[0] != "v2" || parts[1] != "public" {
+		return nil, fmt.Errorf("token does not start with v2.public.")
+	}
+	bytes, err := base64.RawURLEncoding.DecodeString(parts[2])
+	if err != nil {
+		return nil, fmt.Errorf("invalid token body payload: %w", err)
+	}
+	// v2.public tokens have a 64-byte signature after the main body.
+	bytes, err = base64.StdEncoding.DecodeString(string(bytes[:len(bytes)-64]))
+	if err != nil {
+		return nil, fmt.Errorf("invalid token body internal payload: %w", err)
+	}
+	return bytes, nil
+}
+
+func decodeUnsafeReplIdentity(token string) (*api.GovalReplIdentity, error) {
+	bytes, err := decodeUnsafePASETO(token)
+	if err != nil {
+		return nil, err
+	}
+	var replIdentity api.GovalReplIdentity
+	err = proto.Unmarshal(bytes, &replIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("token body not an api.GovalReplIdentity: %w", err)
+	}
+	return &replIdentity, nil
+}
+
+func decodeUnsafeGovalCert(token string) (*api.GovalCert, error) {
+	bytes, err := decodeUnsafePASETO(token)
+	if err != nil {
+		return nil, err
+	}
+	var decodedCert api.GovalCert
+	err = proto.Unmarshal(bytes, &decodedCert)
+	if err != nil {
+		return nil, fmt.Errorf("token body not an api.GovalReplIdentity: %w", err)
+	}
+	return &decodedCert, nil
+}
+
+// DumpToken returns a string representation explaining a token.
```

Comment:
wouldn't `DecodeToken` make more sense?

Reply:
not really, because decoding typically implies that it's returning something in a usable form, but here it's "not" (it's intentionally not machine-readable, for instance).

but maybe `DebugToken` might be a better fit, and also announces that it's only for debugging.

Reply:
all the libraries i've used say "decode". Otherwise it's "verify".

Reply:
but they're also not amazing libs